mlflow/langchain/__init__.py

```diff
@@ -256,7 +258,8 @@ def load_retriever(persist_directory):
                 f"Current code paths: {code_paths}"
             )
 
-    code_dir_subpath = _validate_and_copy_code_paths(formatted_code_path, path)
+    code_dir_subpath = _validate_and_copy_code_paths(code_paths, path)
+    model_code_dir_subpath = _validate_and_copy_model_code_path(model_code_path, path)
 
     if signature is None:
```

Comment:
When the signature is None, I am not sure if this code would work: `_LangChainModelWrapper(lc_model)` 
So we need to figure out a way to load the model here and use that as a wrapped model so we can infer_signature